models/agentProfile.ts

```diff
@@ -0,0 +1,104 @@
+import dbClient from "../lib/db";
+
+export async function add(profiles) {
+  if(!Array.isArray(profiles)) {
+    profiles = [profiles]
+  }
```

Comment:
I don't really like the idea of methods checking and modifying inputs and _accepting_ different types (string and array) for the same param. IMO, it makes the API contract/interface more complex to understand for the user.

Have you considered:
1. Doing the checking and modification of `profiles` into an array before calling `add()`. And always send an array.
2. Replacing `add()` for two methods `addProfile()` and `addProfiles()`. Where `addProfile()` builds an array and then calls `addProfiles()`.

Reply:
good idea! I went with option 1